src/java.base/windows/native/libnio/fs/WindowsNativeDispatcher.c

```diff
@@ -521,8 +521,15 @@ Java_sun_nio_fs_WindowsNativeDispatcher_SetEndOfFile(JNIEnv* env, jclass this,
 {
     HANDLE h = (HANDLE)jlong_to_ptr(handle);
 
-    if (SetEndOfFile(h) == 0)
-        throwWindowsException(env, GetLastError());
+    if (SetEndOfFile(h) == 0) {
+        DWORD error = GetLastError();
+        LARGE_INTEGER size;
+        if (GetFileSizeEx(h, &size) == 0) {
+            throwWindowsException(env, GetLastError());
+        } else if (size.QuadPart != 0) {
+            throwWindowsException(env, error);
+        }
+    }
 }
```

Comment:
The native methods are meant to map to one win32 call where possible. In this case, I think the workaround needs to be in WindowsChannelFile.open.

Reply:
I know. I tried at the Java level first and one cannot verify the size of the NUL device from Java.

Reply:
it's okay to add a new method to WindowsNativeDispatcher if we need it.

Reply:
Added `GetFileSizeEx()`.